compiler/noirc_evaluator/src/acir/mod.rs

```diff
@@ -915,8 +915,11 @@ impl<'a> Context<'a> {
     ) -> usize {
         let return_values = match terminator {
             TerminatorInstruction::Return { return_values, .. } => return_values,
+            TerminatorInstruction::Unreachable { .. } => return 0,
```

Comment:
In what cases do we reach this in acir?
I'm assuming we do since this used to be a catch-all for  `unreachable!`

Reply:
Good question. Yes, I think one of the tests failed so I handled this case. Here's a code that triggers this:

```noir
fn main() -> pub bool {
    assert(false);
    true
}
```

The final SSA is:

```noir
acir(inline) predicate_pure fn main f0 {
  b0():
    constrain u1 0 == u1 1
    unreachable
}
```